src/main/java/com/bootcamp/demo/service/LoginService.java

```diff
@@ -0,0 +1,72 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Session;
+import com.bootcamp.demo.model.User;
+import com.bootcamp.demo.repository.SessionRepository;
+import com.bootcamp.demo.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+
+/**
+ * LoginService implementation
+ * Next to add: forgotPassword()
+ */
+@Service
+public class LoginService implements ILoginService {
+    private UserRepository userRepository;
+    private SessionRepository sessionRepository;
+
+    public LoginService(UserRepository userRepository, SessionRepository sessionRepository) {
+        this.userRepository = userRepository;
+        this.sessionRepository = sessionRepository;
+    }
+
+    /**
+     * @param new_user the new user
+     * @return true, if user was successfully saved, or false if email is already used
+     */
+    @Override
+    public boolean registerUser(User new_user) {
+        if(userRepository.findByEmail(new_user.getEmail())!= null)
+            return false;
+        return userRepository.save(new_user);
+
+    }
+
+    /**
+     * if user was successfully logged in, a new session is created
+     *
+     * @param email    is the email introduced by the user
+     * @param password is the password introduced by the user
+     * @return true if the user with the given email and password exists in the database, else false
+     */
+    @Override
+    public boolean loginUser(String email, String password) {
+        User loggedUser = userRepository.findByEmail(email);
+        if (loggedUser != null && loggedUser.getPassword().equals(password)) {
+            sessionRepository.save(new Session(loggedUser));
+            return true;
+        }
+        return false;
```

Comment:
Same here, let's avoid having multiple return statements, and rewrite the code in such a manner we don't need more than one return to the method.